addons/mrp/models/mrp_production.py

```diff
@@ -333,26 +324,11 @@ def get_stock_moves(moves, state):
                 ('raw_material_production_id.state', 'in', ('cancel', 'done', 'draft'))])
             return [('move_raw_ids', 'in', raw_stock_moves.ids)]
 
-        raw_stock_moves = self.env['stock.move'].search([
-            ('raw_material_production_id', '!=', False),
-            ('raw_material_production_id.state', 'not in', ('cancel', 'done', 'draft'))])
-        if operator == '=':
-            raw_stock_moves = get_stock_moves(raw_stock_moves, value)
-        elif operator == '!=':
-            raw_stock_moves = raw_stock_moves - get_stock_moves(raw_stock_moves, value)
-        elif operator == 'in':
-            search_raw_moves = self.env['stock.move']
-            for state in value:
-                search_raw_moves |= get_stock_moves(raw_stock_moves, state)
-            raw_stock_moves = search_raw_moves
-        elif operator == 'not in':
-            search_raw_moves = self.env['stock.move']
-            for state in value:
-                search_raw_moves |= raw_stock_moves - get_stock_moves(raw_stock_moves, state)
-            raw_stock_moves = search_raw_moves
-        else:
-            raise UserError(_('Operation not supported'))
-        return [('move_raw_ids', 'in', raw_stock_moves.ids)]
+        selected_productions = self.env['mrp.production']
+        for prod in self.env['mrp.production'].search([('state', 'not in', ('done', 'cancel', 'draft'))]):
+            if prod.move_raw_ids._match_searched_availability(operator, value, _get_comparison_date):
+                selected_productions += prod
+        return [('id', 'in', selected_productions.ids)]
```

Comment:
It is not needed to work with a recordset, we can keep the ids instead
```suggestion
        selected_productions_ids = []
        for prod in self.env['mrp.production'].search([('state', 'not in', ('done', 'cancel', 'draft'))]):
            if prod.move_raw_ids._match_searched_availability(operator, value, _get_comparison_date):
                selected_productions_ids.append(prod.id)
        return [('id', 'in', selected_productions_ids)]
```